packages/block-editor/src/components/block-switcher/index.js

```diff
@@ -83,9 +96,21 @@ export const BlockSwitcherDropdownMenu = ( { clientIds, blocks } ) => {
 	const isReusable = blocks.length === 1 && isReusableBlock( blocks[ 0 ] );
 	const isTemplate = blocks.length === 1 && isTemplatePart( blocks[ 0 ] );
 
-	const onTransform = ( name ) =>
+	const onBlockTransform = ( name ) =>
 		replaceBlocks( clientIds, switchToBlockType( blocks, name ) );
+	// TODO comments (for the above too).
+	const onPatternTransform = ( transformedBlocks ) => {
+		// If on replaceMode (currently single Template Part block selected)
```

Comment:
For template parts, I think the replacement should always be of the contents, not the template part container itself.

Reply:
I agree. I have to work on the replacement actions more and test various cases.

Reply:
Wait, do you mean like changing the contents of the selected Template Part? In that case, after saving, the current template part will change everywhere.

Reply:
Yes, correct :) That should be the main flow. Swapping a template part with something else for a specific template can be done by just removing the template part and adding a new pattern or starting from scratch.

Reply:
Implemented in: https://github.com/WordPress/gutenberg/pull/29890/commits/b548e354541ddb40df03ee1fb13f018d05d7811f